app/graphql/queries/authorized_model_query.rb

```diff
@@ -0,0 +1,46 @@
+module Queries
+  class AuthorizedModelQuery
+    def initialize(model_class:, user:)
+      @model_class = model_class
+      @user = user
+    end
+
+    def find_by(params)
```

Comment:
I was starting to write how it should have also `method_missing` and `respond_to?`, but then I realized that it has few problems.

First, the name `find_by` is misleading, because the finding is always on `id`, so it should be `find_by_id` and then `params` can be changed to `id`.

Reply:
Yep, you are right. I'll change this so that it's not too misleading. 👍